### DISCUSSION TITLE
Does the Node fs Module work with Quasar Web App?

### DISCUSSION THREAD

**Discussion by CynthiaBlue** [quasarframework/quasar#11390] [General - Components / Directives / etc]
Is it possible to use the node file system module on a Quasar web app?

The quasar docs say: 

> /quasar.conf.js is run by the Quasar CLI build system, so this code runs under Node directly, not in the context of your app. This means you can require modules like ‘fs’, ‘path’, ‘webpack’ and so on. Make sure the ES6 features that you want to write this file with are supported by the installed version of your Node (which should be >= 10).

[https://quasar.dev/quasar-cli/quasar-conf-js#introduction](https://quasar.dev/quasar-cli/quasar-conf-js#introduction)

But when I try this little test in my code:
```
const fs = require('fs')
    fs.access('test2.vue', err => {
      if (err) {
        return console.error(err)
      }
      console.log(" no Error ")
      //done
    })
```

I get:
```
Module not found: Error: Can't resolve 'fs'
```

According to the Node documentation, nothing needs to be installed because fs comes with node.  I have tried installing fs anyway via npm but I get the same error.

There seems to be a ton of conflicting information online about it.  I would love for my Quasar web app to be able to create directories and files on the server.

I am running Node v16.13.0

Thanks.

**Comment by pdanpdan** [comment]
Where are you writing that code?
What you cited from the docs is explicitly referring to the code in quasar.conf.js, which is only executed when the application is built or when I'm dev mode when the file is changed.

I have never in my life seen any kind of contradictory information on the web about the possibility of an application that runs in the browser on the client computer to change in any way some files on the server.

**Reply by CynthiaBlue** [reply]
Well... I don't know.  I'm writing a web app and I want to use Node fs in Quasar and it's giving me that error.

**Reply by hawkeye64** [reply]
The context of Node is used to build the web app, not run it. The context of a web browser runs your web app. However, there are caveats. You can make a hybrid web app with Electron. In this case, Electron contains both Node (main) and Chromium (renderer).
